### PR TITLE
home-assistant dependencies

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

More home-assistant dep updates

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
The octoprint build was broken on master, due to a version constraint for `watchdog==0.10.4`, and additionally on this branch due to `emoji==0.6.0`. Awesome version constraints ignore infrastructure. Kudos to the octoprint maintainers!

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 111661` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>37 packages built:</summary>
  <ul>
    <li>home-assistant</li>
    <li>ntfy</li>
    <li>octoprint</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.emoji</li>
    <li>python37Packages.pysonos</li>
    <li>python37Packages.pywilight</li>
    <li>python37Packages.xknx</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy</li>
    <li>python37Packages.zigpy-deconz</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.brother</li>
    <li>python38Packages.emoji</li>
    <li>python38Packages.pysonos</li>
    <li>python38Packages.pywilight</li>
    <li>python38Packages.xknx</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy</li>
    <li>python38Packages.zigpy-cc</li>
    <li>python38Packages.zigpy-deconz</li>
    <li>python38Packages.zigpy-xbee</li>
    <li>python38Packages.zigpy-zigate</li>
    <li>python38Packages.zigpy-znp</li>
    <li>python39Packages.bellows</li>
    <li>python39Packages.emoji</li>
    <li>python39Packages.pywilight</li>
    <li>python39Packages.xknx</li>
    <li>python39Packages.zha-quirks</li>
    <li>python39Packages.zigpy</li>
    <li>python39Packages.zigpy-cc</li>
    <li>python39Packages.zigpy-deconz</li>
    <li>python39Packages.zigpy-xbee</li>
    <li>python39Packages.zigpy-zigate</li>
    <li>python39Packages.zigpy-znp</li>
    <li>terraform-compliance</li>
  </ul>
</details>

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 111661` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>51 packages built:</summary>
  <ul>
    <li>beets</li>
    <li>home-assistant</li>
    <li>mpdris2</li>
    <li>ntfy</li>
    <li>octoprint</li>
    <li>poezio</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.emoji</li>
    <li>python37Packages.influxdb-client</li>
    <li>python37Packages.mpd2</li>
    <li>python37Packages.poezio</li>
    <li>python37Packages.pysonos</li>
    <li>python37Packages.pywilight</li>
    <li>python37Packages.xknx</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy</li>
    <li>python37Packages.zigpy-deconz</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.brother</li>
    <li>python38Packages.emoji</li>
    <li>python38Packages.influxdb-client</li>
    <li>python38Packages.mpd2</li>
    <li>python38Packages.pysonos</li>
    <li>python38Packages.pywilight</li>
    <li>python38Packages.xknx</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy</li>
    <li>python38Packages.zigpy-cc</li>
    <li>python38Packages.zigpy-deconz</li>
    <li>python38Packages.zigpy-xbee</li>
    <li>python38Packages.zigpy-zigate</li>
    <li>python38Packages.zigpy-znp</li>
    <li>python39Packages.bellows</li>
    <li>python39Packages.emoji</li>
    <li>python39Packages.influxdb-client</li>
    <li>python39Packages.mpd2</li>
    <li>python39Packages.poezio</li>
    <li>python39Packages.pywilight</li>
    <li>python39Packages.xknx</li>
    <li>python39Packages.zha-quirks</li>
    <li>python39Packages.zigpy</li>
    <li>python39Packages.zigpy-cc</li>
    <li>python39Packages.zigpy-deconz</li>
    <li>python39Packages.zigpy-xbee</li>
    <li>python39Packages.zigpy-zigate</li>
    <li>python39Packages.zigpy-znp</li>
    <li>qtile</li>
    <li>rofi-mpd</li>
    <li>sonata</li>
    <li>terraform-compliance</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 111661` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>51 packages built:</summary>
  <ul>
    <li>beets</li>
    <li>home-assistant</li>
    <li>mpdris2</li>
    <li>ntfy</li>
    <li>octoprint</li>
    <li>poezio (python38Packages.poezio)</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.emoji</li>
    <li>python37Packages.influxdb-client</li>
    <li>python37Packages.mpd2</li>
    <li>python37Packages.poezio</li>
    <li>python37Packages.pysonos</li>
    <li>python37Packages.pywilight</li>
    <li>python37Packages.xknx</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy</li>
    <li>python37Packages.zigpy-deconz</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.brother</li>
    <li>python38Packages.emoji</li>
    <li>python38Packages.influxdb-client</li>
    <li>python38Packages.mpd2</li>
    <li>python38Packages.pysonos</li>
    <li>python38Packages.pywilight</li>
    <li>python38Packages.xknx</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy</li>
    <li>python38Packages.zigpy-cc</li>
    <li>python38Packages.zigpy-deconz</li>
    <li>python38Packages.zigpy-xbee</li>
    <li>python38Packages.zigpy-zigate</li>
    <li>python38Packages.zigpy-znp</li>
    <li>python39Packages.bellows</li>
    <li>python39Packages.emoji</li>
    <li>python39Packages.influxdb-client</li>
    <li>python39Packages.mpd2</li>
    <li>python39Packages.poezio</li>
    <li>python39Packages.pywilight</li>
    <li>python39Packages.xknx</li>
    <li>python39Packages.zha-quirks</li>
    <li>python39Packages.zigpy</li>
    <li>python39Packages.zigpy-cc</li>
    <li>python39Packages.zigpy-deconz</li>
    <li>python39Packages.zigpy-xbee</li>
    <li>python39Packages.zigpy-zigate</li>
    <li>python39Packages.zigpy-znp</li>
    <li>qtile</li>
    <li>rofi-mpd</li>
    <li>sonata</li>
    <li>terraform-compliance</li>
  </ul>
</details>

File: pkgs/applications/misc/octoprint/default.nix [hunk 1]
```diff
@@ -144,7 +144,9 @@ let
 
               postPatch = let
                 ignoreVersionConstraints = [
+                  "emoji"
                   "sentry-sdk"
+                  "watchdog"
```
Comment (fabaff) [inline pkgs/applications/misc/octoprint/default.nix:149]: That's indeed very neat to avoid build failures.

**Review by fabaff — APPROVED** [review]
Looks good to me.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111661` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages marked as broken and skipped:
</summary>
<ul>
<li>poezio</li>
<li>python37Packages.bellows</li>
<li>python37Packages.poezio</li>
<li>python38Packages.bellows</li>
<li>python38Packages.poezio</li>
<li>python39Packages.bellows</li>
<li>python39Packages.poezio</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>ntfy: log <a href="https://termbin.com/hcqp">https://termbin.com/hcqp</a></li>
  </ul>
</details>
<details>
<summary>
24 packages built:
</summary>
<ul>
<li>home-assistant</li>
<li>mpdris2</li>
<li>octoprint</li>
<li>python37Packages.emoji</li>
<li>python37Packages.influxdb-client</li>
<li>python37Packages.mpd2</li>
<li>python37Packages.pysonos</li>
<li>python37Packages.pywilight</li>
<li>python37Packages.xknx</li>
<li>python38Packages.brother</li>
<li>python38Packages.emoji</li>
<li>python38Packages.influxdb-client</li>
<li>python38Packages.mpd2</li>
<li>python38Packages.pysonos</li>
<li>python38Packages.pywilight</li>
<li>python38Packages.xknx</li>
<li>python39Packages.brother</li>
<li>python39Packages.emoji</li>
<li>python39Packages.influxdb-client</li>
<li>python39Packages.mpd2</li>
<li>python39Packages.pywilight</li>
<li>python39Packages.xknx</li>
<li>rofi-mpd</li>
<li>terraform-compliance</li>
</ul>
</details>

**Comment by mweinelt** [comment]
That's it for now.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 111661` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>60 packages built:</summary>
  <ul>
    <li>beets</li>
    <li>home-assistant</li>
    <li>mpdris2</li>
    <li>ntfy</li>
    <li>octoprint</li>
    <li>poezio</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.bt_proximity</li>
    <li>python37Packages.emoji</li>
    <li>python37Packages.fixerio</li>
    <li>python37Packages.influxdb-client</li>
    <li>python37Packages.libsoundtouch</li>
    <li>python37Packages.mpd2</li>
    <li>python37Packages.poezio</li>
    <li>python37Packages.pysonos</li>
    <li>python37Packages.pywilight</li>
    <li>python37Packages.xknx</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy</li>
    <li>python37Packages.zigpy-deconz</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.brother</li>
    <li>python38Packages.bt_proximity</li>
    <li>python38Packages.emoji</li>
    <li>python38Packages.fixerio</li>
    <li>python38Packages.influxdb-client</li>
    <li>python38Packages.libsoundtouch</li>
    <li>python38Packages.mpd2</li>
    <li>python38Packages.pysonos</li>
    <li>python38Packages.pywilight</li>
    <li>python38Packages.xknx</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy</li>
    <li>python38Packages.zigpy-cc</li>
    <li>python38Packages.zigpy-deconz</li>
    <li>python38Packages.zigpy-xbee</li>
    <li>python38Packages.zigpy-zigate</li>
    <li>python38Packages.zigpy-znp</li>
    <li>python39Packages.bellows</li>
    <li>python39Packages.bt_proximity</li>
    <li>python39Packages.emoji</li>
    <li>python39Packages.fixerio</li>
    <li>python39Packages.influxdb-client</li>
    <li>python39Packages.libsoundtouch</li>
    <li>python39Packages.mpd2</li>
    <li>python39Packages.poezio</li>
    <li>python39Packages.pywilight</li>
    <li>python39Packages.xknx</li>
    <li>python39Packages.zha-quirks</li>
    <li>python39Packages.zigpy</li>
    <li>python39Packages.zigpy-cc</li>
    <li>python39Packages.zigpy-deconz</li>
    <li>python39Packages.zigpy-xbee</li>
    <li>python39Packages.zigpy-zigate</li>
    <li>python39Packages.zigpy-znp</li>
    <li>qtile</li>
    <li>rofi-mpd</li>
    <li>sonata</li>
    <li>terraform-compliance</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111661` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
10 packages marked as broken and skipped:
</summary>
<ul>
<li>poezio</li>
<li>python37Packages.bellows</li>
<li>python37Packages.bt_proximity</li>
<li>python37Packages.poezio</li>
<li>python38Packages.bellows</li>
<li>python38Packages.bt_proximity</li>
<li>python38Packages.poezio</li>
<li>python39Packages.bellows</li>
<li>python39Packages.bt_proximity</li>
<li>python39Packages.poezio</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>ntfy: log <a href="https://termbin.com/jddr">https://termbin.com/jddr</a></li>
  </ul>
</details>
<details>
<summary>
30 packages built:
</summary>
<ul>
<li>home-assistant</li>
<li>mpdris2</li>
<li>octoprint</li>
<li>python37Packages.emoji</li>
<li>python37Packages.fixerio</li>
<li>python37Packages.influxdb-client</li>
<li>python37Packages.libsoundtouch</li>
<li>python37Packages.mpd2</li>
<li>python37Packages.pysonos</li>
<li>python37Packages.pywilight</li>
<li>python37Packages.xknx</li>
<li>python38Packages.brother</li>
<li>python38Packages.emoji</li>
<li>python38Packages.fixerio</li>
<li>python38Packages.influxdb-client</li>
<li>python38Packages.libsoundtouch</li>
<li>python38Packages.mpd2</li>
<li>python38Packages.pysonos</li>
<li>python38Packages.pywilight</li>
<li>python38Packages.xknx</li>
<li>python39Packages.brother</li>
<li>python39Packages.emoji</li>
<li>python39Packages.fixerio</li>
<li>python39Packages.influxdb-client</li>
<li>python39Packages.libsoundtouch</li>
<li>python39Packages.mpd2</li>
<li>python39Packages.pywilight</li>
<li>python39Packages.xknx</li>
<li>rofi-mpd</li>
<li>terraform-compliance</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.fixerio:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern 'requests==2.10.0' doesn't match anything in file 'setup.py'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>
python38Packages.fixerio:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern 'requests==2.10.0' doesn't match anything in file 'setup.py'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>
python39Packages.fixerio:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern 'requests==2.10.0' doesn't match anything in file 'setup.py'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>

**Comment by mweinelt** [comment]
> substituteStream(): WARNING: pattern 'requests==2.10.0' doesn't match anything in file 'setup.py'

removed.

**Pull request merged**